src/index.js

```diff
@@ -16,7 +16,8 @@ var enhance = module.exports = function enhance(proto) {
 
   var listeners = new WeakMap()
   proto.addEventListener = function(name, originalCallback, optionsOrCapture) {
-    if (optionsOrCapture === undefined || optionsOrCapture === true || optionsOrCapture === false) {
+    var expectedOriginalCallback = Boolean(originalCallback && typeof originalCallback === 'function' || typeof originalCallback === 'object')
+    if (!expectedOriginalCallback || optionsOrCapture === undefined || optionsOrCapture === true || optionsOrCapture === false) {
```

Comment:
Could the condition be rewritten as:

```js
if (optionsOrCapture === undefined || optionsOrCapture === true || optionsOrCapture === false || !Boolean(originalCallback && typeof originalCallback === 'function' || typeof originalCallback === 'object'))
```

That way the `typeof` checks are avoided in the happy path where the `optionsOrCapture === undefined` short-cirtuits the rest of the expression.

Reply:
Rewrote it, but the linter doesn't like `!Boolean(...)` so I had to tweak the conditions a little bit.